### PR TITLE
Add prune operation

### LINKED ISSUE #8: Rename `gc` to something else
We should rename `gc` operation to something else to avoid confustion with `ipfs gc` command.

How about `prune` ?

[Screenshot_2019-10-24 prune - Wiktionary](https://en.wiktionary.org/wiki/prune#Etymology_2)

### PR DESCRIPTION
**Pull request by hacdias**
Close #8.

### CODE REVIEW

File: SPEC.md [hunk 1]
```diff
@@ -127,3 +128,19 @@ $ ipfs files mv /cohosting/lazy/ipfs.io /cohosting/full/ipfs.io
 ```
 
 Or the other way around.
+
+### Prune
```
Comment (lidel) [inline SPEC.md]:
```suggestion
### Pruning
```

File: SPEC.md [hunk 2]
```diff
@@ -18,6 +18,7 @@
   * [Listing](#listing)
   * [Updating](#updating)
   * [Changing cohosting type](#changing-cohosting-type)
+  * [Prune](#prune)
```
Comment (lidel) [inline SPEC.md]:
```suggestion
  * [Pruning](#pruning)
```

File: SPEC.md [hunk 3]
```diff
@@ -127,3 +128,19 @@ $ ipfs files mv /cohosting/lazy/ipfs.io /cohosting/full/ipfs.io
...
 
 Or the other way around.
+
+### Prune
+
+The act of deleting old snapshots. To remove all snapshots for a certain domain:
+
+```console
+$ ipfs files rm -r /cohosting/<lazy|full>/ipfs.io
+```
```
Comment (lidel) [inline SPEC.md]: I would remove this, as it looks the same as `Removing` section

File: SPEC.md [hunk 4]
```diff
@@ -127,3 +128,19 @@ $ ipfs files mv /cohosting/lazy/ipfs.io /cohosting/full/ipfs.io
...
+
+To remove all but the last _n_ snapshots:
```
Comment (lidel) [inline SPEC.md]:
```suggestion
Pruning is the act of deleting some of oldest snapshots while keeping _n_ recent ones around.
```

File: SPEC.md [hunk 5]
```diff
@@ -127,3 +128,19 @@ $ ipfs files mv /cohosting/lazy/ipfs.io /cohosting/full/ipfs.io
...
+
+```console
+$ ipfs files ls /cohosting/<lazy|full>/ipfs.io | tail -r | tail -n +3
+# For each snapshot
+$ ipfs files ls /cohosting/<lazy|full>/ipfs.io/$snapshot
+```
```
Comment (lidel) [inline SPEC.md]: Let's make example specific and platform-agnostic (`tail -r` is not valid on Linux):

```bash
# assuming we have 3 snapshots:
$ ipfs files ls /cohosting/full/docs.ipfs.io | sort
2019-10-10_000001
2019-10-11_000002
2019-10-12_000003

# we decide to keep only the last (latest) one (n=1)
# 1. get snapshots ready to be pruned:
$ ipfs files ls /cohosting/full/docs.ipfs.io | sort | head -n -1
2019-10-10_000001
2019-10-11_000002

# 2. perform pruning by removing each snapshot:
$ ipfs files rm -r /cohosting/full/ipfs.io/2019-10-10_000001
$ ipfs files rm -r /cohosting/full/ipfs.io/2019-10-11_000002
```

**Review by lidel — CHANGES_REQUESTED** [review]

File: SPEC.md (hunk 5 above)
Comment (hacdias) [reply SPEC.md]: THANKS! I was actually going to ask you help on this one, but then I found out about `tail -r` and thought it would work everywhere!

**Comment by hacdias** [comment]
@lidel just updated as per your recommendations!

**Comment by hacdias** [comment]
I'm merging this as all requested changes were made.

**Pull request merged**